src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/ItemManager.cs

```diff
@@ -229,39 +231,60 @@ internal ItemManager(RecentItemsManager recentItemsManager, IGlobalOptionService
-                }
-                else
-                {
-                    Contract.ThrowIfNull(document);
-                    filterMethod = (itemsWithPatternMatches, text) => completionService.FilterItems(document, itemsWithPatternMatches, text);
+                    var tick = Environment.TickCount;
+
+                    var selectedItem = initialListOfItemsToBeIncluded[selectedItemIndex].RoslynCompletionItem;
+
+                    // "Preselect" is only used when we have high confidence with the selection, so don't override it.
+                    if (selectedItem.Rules.MatchPriority < MatchPriority.Preselect)
+                    {
+                        int defaultsMatchingIndex;
+                        var useAggressiveDefaultsMatching = session.TextView.Options.GetOptionValue<bool>(AggressiveDefaultsMatchingOptionName);
+
+                        if (useAggressiveDefaultsMatching)
+                        {
+                            defaultsMatchingIndex = GetAggressiveDefaultsMatch(initialListOfItemsToBeIncluded, data.Defaults);
+                            if (!hasSuggestedItemOptions && defaultsMatchingIndex >= 0)
+                                selectionHint = UpdateSelectionHint.Selected;
+                        }
+                        else
+                        {
+                            defaultsMatchingIndex = GetDefaultsMatch(filterText, initialListOfItemsToBeIncluded, selectedItemIndex, data.Defaults);
+                        }
+
+                        if (defaultsMatchingIndex >= 0)
+                        {
+                            selectedItemIndex = defaultsMatchingIndex;
+                        }
+                    }
+
+                    AsyncCompletionLogger.LogGetDefaultsMatchTicksDataPoint(Environment.TickCount - tick);
                 }
 
-                return HandleNormalFiltering(
-                    filterMethod,
-                    filterText,
+                var showCompletionItemFilters = _globalOptions.GetOption(CompletionViewOptions.ShowCompletionItemFilters, document?.Project.Language);
+                var updatedFilters = showCompletionItemFilters
+                    ? GetUpdatedFilters(initialListOfItemsToBeIncluded, data.SelectedFilters)
+                    : ImmutableArray<CompletionFilterWithState>.Empty;
+
+                return new FilteredCompletionModel(
+                    items: GetHighlightedList(initialListOfItemsToBeIncluded, filterText, highlightMatchingPortions, completionHelper),
+                    selectedItemIndex,
                     updatedFilters,
-                    filterReason,
-                    data.Trigger.Character,
-                    initialListOfItemsToBeIncluded,
-                    hasSuggestedItemOptions,
-                    highlightMatchingPortions,
-                    completionHelper);
+                    selectionHint,
+                    centerSelection: true,
```

Comment:
what does 'centerSelection' mean?

Reply:
this was not answered.

Reply:
"Whether selected item should be displayed in the center of the list. Usually, this is true" from the [comment](http://index/?query=centerse&rightProject=Microsoft.VisualStudio.Language&file=AsyncCompletion%2FData%2FFilteredCompletionModel.cs&line=35)